pypeit/coadd2d.py

```diff
@@ -169,39 +173,114 @@ def __init__(self, spec2d, spectrograph, par, det=1, offsets=None, weights='auto
         self.show_peaks = show_peaks
         self.debug_offsets = debug_offsets
         self.debug = debug
+        self.offsets = None
         self.stack_dict = None
         self.pseudo_dict = None
 
         self.objid_bri = None
         self.slitid_bri  = None
         self.snr_bar_bri = None
+        self.use_weights = None
+        self.wave_grid = None
+        self.good_slits = None
+        self.maskdef_offset = None
 
 
         # Load the stack_dict
         self.stack_dict = self.load_coadd2d_stacks(self.spec2d)
         self.pypeline = self.spectrograph.pypeline
 
+        self.nexp = len(self.spec2d)
+
         # Check that there are the same number of slits on every exposure
         nslits_list = [slits.nslits for slits in self.stack_dict['slits_list']]
-        if not len(set(nslits_list))==1:
+        if not len(set(nslits_list)) == 1:
             msgs.error('Not all of your exposures have the same number of slits. Check your inputs')
-        # TODO: Do the same check above but for the shape and binning
-        # of the input images?
-        self.nslits = nslits_list[0]
-        self.nexp = len(self.stack_dict['specobjs_list'])
+        # This is the number of slits of the single (un-coadded) frames
+        self.nslits_single = nslits_list[0]
+
+        # Check that nspec is the same for all the exposures
+        nspec_list = [slits.nspec for slits in self.stack_dict['slits_list']]
```

Comment:
This variable is exactly the same as nslits_list define above. You have replicated that line and defined a new variable.